.github/workflows/pr.yml

```diff
@@ -5,59 +5,79 @@ on:
     branches: [ main ]
 
 env:
-  CARGO_TERM_COLOR: always
+  CARGO_INCREMENTAL: 1
 
 jobs:
-  test-and-lint:
+  test-linux:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout source
         uses: actions/checkout@v3
 
-      - uses: Swatinem/rust-cache@v2
```

Comment:
why? cache is good for pr

Reply:
Because rust-cache doesn't find any cache.
Example:
- https://github.com/yozhgoor/cargo-temp/actions/runs/4680082221/jobs/8290965377
- https://github.com/yozhgoor/cargo-temp/actions/runs/4679980783/jobs/8290731915
- https://github.com/yozhgoor/cargo-temp/actions/runs/4048717033/jobs/6964293649
- https://github.com/yozhgoor/cargo-temp/actions/runs/4026245369/jobs/6920510401

In each of these jobs you can see something like:
```console
... Restoring cache ...
No cache found.
```

Reply:
In my projects I usually cache on failure too:
```
      - uses: Swatinem/rust-cache@v2
        with:
          cache-on-failure: true
```
I don't know why it's not the default...